sdk/search/search-documents/src/searchServiceClient.ts

```diff
@@ -163,6 +170,14 @@ export class SearchServiceClient {
     this.client = new GeneratedClient(dummyCredential, this.apiVersion, this.endpoint, pipeline);
   }
 
+  /**
+   * Retrieves the search index client corresponding to this search service client
+   * @param indexName Name of the index
+   */
+  public getSearchIndexClient<T>(indexName: string): SearchIndexClient<T> {
+    return new SearchIndexClient<T>(this.endpoint, indexName, this.credential, this.options);
```

Comment:
I feel like we should allow the user to pass custom client options to this method. I'm fine with defaulting to the same options as the service client if none are specified.

Reply:
Done